src/helpers/daily-data-providers/combined-sleep-v2.ts

```diff
@@ -1,39 +1,80 @@
-import { add } from 'date-fns';
-import { totalSleepMinutes as fitbitTotalSleepMinutesDataProvider } from './fitbit-sleep-v2'
-import { totalSleepMinutes as garminTotalSleepMinutesDataProvider } from './garmin-sleep-v2'
-import { totalSleepMinutes as appleHealthSleepMinutesDataProvider } from './apple-health-sleep-v2';
-import getDayKey from '../get-day-key';
-import MyDataHelps from '@careevolution/mydatahelps-js';
+import { add } from "date-fns";
+import { totalSleepMinutes as fitbitTotalSleepMinutesDataProvider } from "./fitbit-sleep-v2";
+import { totalSleepMinutes as garminTotalSleepMinutesDataProvider } from "./garmin-sleep-v2";
+import { totalSleepMinutes as appleHealthSleepMinutesDataProvider } from "./apple-health-sleep-v2";
+import { totalSleepMinutes as healthConnectTotalSleepMinutesDataProvider } from "./health-connect-sleep";
+import getDayKey from "../get-day-key";
+import { getCombinedDataCollectionSettings } from "./combined-data-collection-settings";
 
-export default async function (startDate: Date, endDate: Date): Promise<{ [key: string]: number }> {
-    let providers: Promise<{ [key: string]: number }>[] = [];
+export default async function (
+    startDate: Date,
+    endDate: Date,
+): Promise<Record<string, number>> {
+    const providers: Promise<Record<string, number>>[] = [];
+
+    const combinedSettings = await getCombinedDataCollectionSettings();
+    const { settings, deviceDataV2Types } = combinedSettings;
 
-    const settings = await MyDataHelps.getDataCollectionSettings();
     if (settings.fitbitEnabled) {
         providers.push(fitbitTotalSleepMinutesDataProvider(startDate, endDate));
     }
+
     if (settings.garminEnabled) {
         providers.push(garminTotalSleepMinutesDataProvider(startDate, endDate));
     }
-    if (settings.queryableDeviceDataTypes.find(s => s.namespace == "AppleHealth" && s.type == "SleepAnalysisInterval")) {
+
+    if (
+        settings.appleHealthEnabled &&
+        settings.queryableDeviceDataTypes.some(
+            (ddt) =>
+                ddt.namespace === "AppleHealth" &&
+                ddt.type === "SleepAnalysisInterval",
+        )
+    ) {
         providers.push(appleHealthSleepMinutesDataProvider(startDate, endDate));
     }
 
-    const results = providers.length ? await Promise.all(providers) : [];
+    if (
+        settings.healthConnectEnabled &&
+        deviceDataV2Types.some(
+            (ddt) =>
+                ddt.enabled &&
+                ddt.namespace === "HealthConnect" &&
+                ddt.type === "sleep",
+        )
+    ) {
+        providers.push(
+            healthConnectTotalSleepMinutesDataProvider(startDate, endDate),
+        );
+    }
+
+    if (!providers.length) {
+        return {};
+    }
+
+    const results = await Promise.all(providers);
+    const data: Record<string, number> = {};
 
-    let data: { [key: string]: number; } = {};
-    let currentDate = startDate;
+    let currentDate = new Date(startDate);
```

Comment:
Curious why we need a new date here?

Reply:
Doesn't hurt to have this, but `date-fns` never modifies the dates.  It always returns new ones.  So `currentDate` just gets re-assigned below without any modification to the original.